### PR TITLE
Fixed illegal optimizer rule on multiple consecutive SORTs (3.8)

### PR DESCRIPTION
**Pull request by hkernbach**
Backport of: https://github.com/arangodb/arangodb/pull/16718
Please see all notes there.

### CODE REVIEW

File: CHANGELOG [hunk 1]
```diff
@@ -408,12 +411,12 @@ v3.8.3 (2021-11-17)
   replication operations.
 
 * Close a potential gap during shard synchronization when moving from the
-  initial sync step to the WAL tailing step. In this small gap the leader 
+  initial sync step to the WAL tailing step. In this small gap the leader
```
Comment (mchacki) [inline CHANGELOG:414]: Can we avoid rewriting the full CHANGELOG file?

**Review by mchacki — CHANGES_REQUESTED** [review]
Code good.

Somehowlarge parts of the changeling are changed, I would like to void this and just have the new entry.

File: CHANGELOG (hunk 1 above)
Comment (hkernbach) [reply CHANGELOG]: done

**Review by jsteemann — APPROVED** [review]
LGTM!

**Review by mchacki — APPROVED** [review]
lgtm

**Pull request merged**